### PR TITLE
Feature/remove identity royalty module

### PR DESCRIPTION
**Pull request by iamyulong**
## Summary

Update `Identity` blueprint to not create royalty module for new instances.

## Testing

New tests added

### CODE REVIEW

File: radix-engine/src/updates/cuttlefish.rs [hunk 1]
```diff
@@ -466,3 +478,121 @@ fn generate_cuttlefish_update_min_rounds_per_epoch<S: SubstateDatabase + ?Sized>
...
+/// A quick macro for encoding and unwrapping.
+macro_rules! scrypto_encode {
+    (
+        $expr: expr
+    ) => {
+        ::radix_common::prelude::scrypto_encode($expr).unwrap()
+    };
+}
+
+fn generate_cuttlefish_update_identity_to_not_create_royalty_module<
```
Comment (0xOmarA) [inline radix-engine/src/updates/cuttlefish.rs:491]: Very minor: for code consistency, maybe we should refactor this to be similar to our new API for constructing state updates: https://github.com/radixdlt/radixdlt-scrypto/blob/feature/account-getters/radix-engine/src/updates/cuttlefish.rs#L223-L360

**Review by 0xOmarA — APPROVED** [review]

File: radix-engine/src/updates/cuttlefish.rs [hunk 2]
```diff
@@ -20,6 +20,8 @@ pub struct CuttlefishSettings {
     /// updates the min number of rounds per epoch.
     pub update_number_of_min_rounds_per_epoch:
         UpdateSetting<UpdateNumberOfMinRoundsPerEpochSettings>,
+    /// Update identity blueprint to not create royalty module.
```
Comment (dhedey) [inline radix-engine/src/updates/cuttlefish.rs:23]: MINOR: Slight typo: `Update identity blueprint to not create a royalty module`

File: radix-engine/src/updates/cuttlefish.rs [hunk 3]
```diff
@@ -466,3 +478,121 @@ fn generate_cuttlefish_update_min_rounds_per_epoch<S: SubstateDatabase + ?Sized>
...
+        ::radix_common::prelude::scrypto_encode($expr).unwrap()
+    };
+}
+
+fn generate_cuttlefish_update_identity_to_not_create_royalty_module<
+    S: SubstateDatabase + ?Sized,
+>(
+    db: &S,
+) -> StateUpdates {
+    let reader = SystemDatabaseReader::new(db);
```
Comment (dhedey) [inline radix-engine/src/updates/cuttlefish.rs:496]: Do we also want to consider updating all the TYPE_INFO substates for _existing_ Identities to disable their royalty modules? (assuming it wouldn't break Gateway)

**Review by dhedey — APPROVED** [review]

File: radix-engine/src/updates/cuttlefish.rs (hunk 3 above)
Comment (iamyulong) [reply radix-engine/src/updates/cuttlefish.rs:496]: I thought about this - to remove all the substates as a result of the royalty module. 

I felt it's too big a change and it's safer to do it post-cuttlefish (when we have a known list of identities with royalty module).

Comment (dhedey) [reply radix-engine/src/updates/cuttlefish.rs:496]: Makes sense to me

File: radix-engine/src/updates/cuttlefish.rs (hunk 1 above)
Comment (dhedey) [reply radix-engine/src/updates/cuttlefish.rs:491]: May as well fix this when we merge develop :)

**Pull request merged**